services/QuillLMS/app/services/demo/create_admin_report.rb

```diff
@@ -64,8 +76,9 @@ def data
       classroom = Classroom.create(name: row['Classroom'])
       all_classrooms.push(classroom)
       ClassroomsTeacher.create(classroom: classroom, user: teacher, role: ClassroomsTeacher::ROLE_TYPES[:owner])
-      student_names = (1..5).to_a.map { |i| "#{Faker::Name.first_name} #{Faker::Name.last_name}" }
+      student_names = (1..25).to_a.map { |i| "#{Faker::Name.first_name} #{Faker::Name.last_name}" }
       Demo::ReportDemoCreator.create_demo_classroom_data(teacher, is_teacher_demo: false, classroom: classroom, student_names: student_names)
+      sleep @delay
```

Comment:
Hey @anathomical -- so this is a good point, and I tried to rewrite it your way, but some other issues came up that made testing it difficult and since there is some time sensitivity here I think it's better to just get it up the way it is and rewrite in the future if necessary. Specifically, I started getting Redis:OOO memories again on staging (the issue the batch delay was added to correct), but because each instance of the demo creation was a worker, this resulted in a bajillion concept result workers getting created again on every retry. I think there are various ways around this, including adding a new queue like you suggested, but for now I think we should leave as is.

Reply:
@emilia-friedberg That seems reasonable to me.